lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
 
-    component = getMountedComponent({ variant: 'secondary' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'secondary');
+    rerender(<NxDropdown {...minimalProps} variant="secondary" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--secondary');
 
-    component = getMountedComponent({ variant: 'error' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'error');
+    rerender(<NxDropdown {...minimalProps} variant="error" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--error');
   });
 
   it('correctly renders the menu based on isOpen prop', function() {
-    let component = getMountedComponent({ isOpen: true });
-    expect(component.find(NxDropdownMenu)).toExist();
+    const { container, rerender } = quickRender({ ...minimalProps, isOpen: true });
+    const menu = container.querySelector('.nx-dropdown-menu');
 
-    component = getMountedComponent({ isOpen: false });
-    expect(component.find(NxDropdownMenu)).not.toExist();
+    expect(menu).toBeInTheDocument();
+    rerender(<NxDropdown {...minimalProps} isOpen={false} />);
+    expect(menu).not.toBeInTheDocument();
   });
 
-  it('correctly assigns supplied classes', function() {
-    const component = getShallowComponent({ className: 'class1 class2' });
-    expect(component).toHaveClassName('nx-dropdown class1 class2');
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
-  it('disables the button (and the toggle fn) when the disabled prop is supplied', function() {
-    const toggleFn = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse: toggleFn, disabled: true }),
-        button = component.find(NxButton);
+  it('disables the button (and the toggle fn) when the disabled prop is supplied', async function() {
+    const user = userEvent.setup();
+    const toggleFn = jest.fn();
+    renderEl({ onToggleCollapse: toggleFn, disabled: true });
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('disabled');
+
+    await user.click(button);
 
-    expect(button).toHaveClassName('disabled');
-    button.simulate('click');
     expect(toggleFn).not.toHaveBeenCalled();
   });
 
-  it('uses VDOM as label if supplied', function() {
-    const label = <NxFontAwesomeIcon icon={faTrash} />,
-        component = getMountedComponent({ label });
-    const button = component.find(NxButton);
+  it('renders react element as label if supplied', function() {
+    const label = <NxFontAwesomeIcon icon={faTrash} />;
+    const { container } = quickRender({ label });
 
-    expect(button.childAt(0)).toContainReact(label);
+    expect(container.querySelector('.nx-icon.fa-trash')).toBeInTheDocument();
```

Comment:
These classes are impl details of `NxFontAwesomeIcon`.  A better approach would be something like
```
const view = quickRender({ label: <span data-testid="foo" /> }),
    toggle = view.getByRole('button');
    
expect(within(toggle).getByTestId('foo')).toBeInTheDocument();
```

Reply:
> These classes are impl details of `NxFontAwesomeIcon`. A better approach would be something like
> 
> ```
> const view = quickRender({ label: <span data-testid="foo" /> }),
>     toggle = view.getByRole('button');
>     
> expect(within(toggle).getByTestId('foo')).toBeInTheDocument();
> ```
> 
> This also tests that the label ends up within the toggle button as opposed to somewhere else

Thanks, I updated it here: d227d8a9d808f9062c6dfd90d74a02982194c268